identity/src/main/java/com/shotcutter/identity/user/UserIdentityService.java

```diff
@@ -37,4 +51,39 @@ public UserIdentityService(UserRepository userRepository,
                 .map(userRepository::save);
     }
 
+    public Optional<UserEntity> updateAvatar(String userId, MultipartFile avatar) throws IOException {
+        var path = getUserAvatarPath(userId, avatar);
+        var user = findById(userId).get();
+
+        // if avatar for this user already exist at s3 - we need to remove it at first
+        if (doesCurrentUserHaveAvatar(user)) {
+            var currentAvatarPath = new URL(user.getAvatar()).getPath();
+            amazonS3.deleteObject(s3BucketName, currentAvatarPath.substring(1));
```

Comment:
please consider async client. I think it's affordable for user not to have avatar updated immediately after getting the response from BE